src/services/words.ts

```diff
@@ -27,14 +30,28 @@ const getByLanguageAndUser = async function(languageId: string, userId: number):
   return result.rows.map((dbItem: WordDB) => convertWordTypes(dbItem));
 };
 
-
-const getUserwordsInText = async function(userId: number, textId: number, simple: boolean = true): Promise<Array<Word>> {
-  const result: QueryResult = await wordData.getUserwordsInText(userId, textId, simple);
-
-  return result.rows.map((dbItem: WordDB) => convertWordTypes(dbItem));
+const getUserwordsInText = async function(userId: number, textId: number, simple: boolean = true): Promise<Array<WordWithTransContextStatusTypes>> {
```

Comment:
I think it may be possible to collect all this data from the database in a single QueryResult using `JOIN`s and `array_agg` (or `string_agg`).

Reply:
Agreed, using a JOIN would be much more efficient if possible.